drivers/rtc/Kconfig

```diff
@@ -2,7 +2,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig RTC
-	bool "Real-Time Clock (RTC) drivers"
+	bool "Real-Time Clock (RTC) drivers [EXPERIMENTAL]"
```

Comment:
Device driver APIs are not experimental features, they follow the lifecycle described [here](https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#api-lifecycle) The RTC device driver API is at this time `unstable` by the actual descriptions used for device driver APIs. This is therefore an inappropriate change.

Reply:
Yeah, I wouldn't change the spinlock API in this pr. If the api is experimental or not, it's kind of irrelevant.

Reply:
@cfriedt That spinlock stuff is now merged in main, so this PR is no longer a draft. The question I asked above wasn't so much about the spinlock (which I never intended to be part of this PR) but about marking the RTC driver API experimental. I'm a bit confused, because it is shown as experimental in the [API overview](https://docs.zephyrproject.org/latest/develop/api/overview.html) plus experimental API drivers are usually marked as such in Kconfig to avoid confusion and (more importantly) flagged as `select EXPERIMENTAL` so they produce automated warnings when used which you confirmed in your first review. @bjarki-trackunit  disagreed on that point, that's why I asked you both to quickly co-ordinate, so that I know what the final consensus will be on that point and finalize this PR then.

Reply:
Clarified with cfried that this change will be removed. The corresponding ticket was closed as "won't fix".